### ISSUE TITLE
Resteasy Reactive Client with multipart forms

### ISSUE THREAD

**Issue by andlinger** [quarkusio/quarkus#22658]
### Description

We use the Resteasy Rest Client with Multipart Forms. We would like to switch to Resteasy Reactive with the whole project. However, in our POJO form we use a multipart annotation for which there is no counterpart in Resteasy Reactive.

It would be great if the `@PartFilename` annotation could be implemented.

```java
public class OurForm {
  @FormParam("files")
  @PartFilename("document.txt")
  @PartType(MediaType.TEXT_PLAIN)
  private String documentFile;
}
```

**Comment by geoand** [comment]
This is one for @michalszynkiewicz

**Comment by andlinger** [comment]
I'm not quite sure what you mean by server side.

I have following rest client in my Quarkus backend:

```java
@Path("/convert")
@RegisterRestClient
public interface Client {

    @POST
    @Consumes(MediaType.MULTIPART_FORM_DATA)
    @Produces("application/pdf")
    byte[] convert(@MultipartForm OurForm form) throws PdfGenerationException;
}
```

This client is used in our Quarkus backend to send a text file, which we have in our `OurForm` object, to another server and have this text file converted to a PDF.

**Comment by geoand** [comment]
Okay, so understood this wrong.

@michalszynkiewicz this is for you after all

**Comment by michalszynkiewicz** [comment]
I added this to my todo list but I won't have time for it soon so I'm leaving it unassigned.

**Comment by geoand** [comment]
I'll take it as it seems like something we want for `2.7`

**Comment by andlinger** [comment]
@geoand Are you sure this works like the non-reactive rest client? I just changed the dependencies to the Reactive Rest Client on Quarkus 2.7.0.CR1 and now it doesn't work anymore. The server, which is contacted with the REST client, no longer recognizes the transferred data.

Update:
The @PartFilename annotation only works when the annotated member is a java.io.File object. In the past this also worked if the member was of type String.

**Comment by geoand** [comment]
Right, we should probably only allow it on `File`, which is semantically more correct. WDYT @michalszynkiewicz ?

**Comment by andlinger** [comment]
But in the JAX-RS client it also works with String and Buffer. Shouldn't it be the same in both implementations?

**Comment by geoand** [comment]
There is always a case to be made whether we should be perfectly compatible vs doing something that more semantically correct.

When it involves non-core features, I tend to prefer the latter. But in this case, it's ultimately up to @michalszynkiewicz

**Comment by andlinger** [comment]
Is there a decision on this?
